Fix issue preventing DynamoDB stream consumers to start from `LATEST`

dannycranmer opened the pull request:
*Issue #, if available:*
https://github.com/awslabs/amazon-kinesis-connector-flink/issues/9

*Description of changes:*
Fix issue preventing DynamoDB stream consumers to start from `LATEST` 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

aalexandrov reviewed (approved):
Some minor comments (see below). I also suggest to inline the logic currently defined in 
```
software.amazon.kinesis.connectors.flink.util.AWSUtil#getStartingPosition
```
in the call site
```
software.amazon.kinesis.connectors.flink.internals.KinesisDataFetcher#createRecordPublisher
``` 
for better symmetry with its counterpart
```
software.amazon.kinesis.connectors.flink.internals.DynamoDBStreamsDataFetcher#createRecordPublisher
```

You can also remove the overwritten method definition at

```
software.amazon.kinesis.connectors.flink.internals.DynamoDBStreamsDataFetcher#createShardConsumer
```
as the one at
```
software.amazon.kinesis.connectors.flink.internals.KinesisDataFetcher#createShardConsumer
```
is the same.

dannycranmer commented:
Thanks for the comments @aalexandrov. I have removed the duplicated overridden method and moved the factory instantiation to the constructor. Have left the `AWSUtils::getStartingPosition` in place to stop further bloating the scope of `KinesisDataFetcher`
